lib/grape-swagger/doc_methods/parse_params.rb

```diff
@@ -70,7 +70,7 @@ def document_add_extensions(settings)
 
         def document_array_param(value_type, definitions)
           if value_type[:documentation].present?
-            param_type = value_type[:documentation][:param_type]
+            param_type = value_type[:documentation][:param_type] || value_type[:documentation][:in]
```

Comment:
This seems to be the root cause of problems in recognizing a "body"/"formData" type. Currently, it's possible to mark a parameter as a body parameter with any of these options.